File: easy/problems/0504-base-7/solution.ts
function convertToBase7(num: number): string {
    if (num === 0) {
        return "0";
    }

    const sign = num < 0 ? "-" : "";
    let value = Math.abs(num);
    const digits: string[] = [];
    while (value > 0) {
        digits.push(String(value % 7));
        value = Math.floor(value / 7);
    }
    digits.reverse();
    return sign + digits.join("");
}

File: easy/problems/0504-base-7/solution.py
class Solution:
    def convertToBase7(self, num: int) -> str:
        if num == 0:
            return "0"

        sign = "-" if num < 0 else ""
        num = abs(num)
        digits = []
        while num > 0:
            digits.append(str(num % 7))
            num //= 7
        return sign + "".join(reversed(digits))
